corehq/apps/users/forms.py

```diff
@@ -417,11 +415,7 @@ def __init__(self, project, user_id, **kwargs):
         )
 
     def clean_new_password1(self):
-        password1 = decode_password(self.cleaned_data.get('new_password1'))
-        if password1 == '':
-            raise ValidationError(
-                _("Password cannot be empty"), code='new_password1_empty',
-            )
```

Comment:
I this check done somewhere else? I'm surprised that this was part of password obfuscation.

Reply:
Hey @millerdev 

User cannot submit a form with empty password (there can be ways). This check was added in case the decode_password method returns a blank password in rare cases when obfuscation would not be as expected. 
this was added https://github.com/dimagi/commcare-hq/pull/15490/files#diff-68f388f9cbc609677f66c12e41cf02b8R17-R18 and then in this class.